src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -151,6 +155,36 @@ export default App.extend({
 
     this.showList();
   },
+  subscribe() {
+    const channel = Radio.channel('ws');
+    const filterKey = this.getState().getType();
+    const flowStates = this.getState('flowStates');
+
+    channel.request('subscribe', this.collection.models, { filters: { [filterKey]: this.filters } });
+
+    this.listenTo(channel, 'message', ({ category, resource, payload }) => {
+      if (!category) return;
+
+      const modelResource = category === 'ResourceCreated' ? payload.resource : resource;
+      const model = Radio.request('entities', 'get:store', modelResource);
+
+      if (this.collection.get(model) || model.isFetching()) return;
+
+      model.fetch().then(() => {
+        // notifications do not fully filter our action flow_states
+        if (filterKey === 'actions' && !flowStates.includes(model.getFlow().getState().id)) return;
+
+        this.collection.add(model);
+        channel.request('add', model);
+
+        if (model.notifications) {
```

Comment:
we should either entirely handle this inside the base model, or entirely handle it here..  perhaps the base model is better..  I can't think of why we wouldn't always want this to happen

Reply:
I think it would work to do this in the base model.

But does `this.collection.add(model)` need to happen before we handle the queued notifications? Or does that order not matter?

For example if the order of things happened like this, would the correct flow.state show in the UI?

1. A `StateChanged` notification is queued while a new flow is fetched.
2. Flow fetch finishes.
3. Queued `StateChanged` ws notification is handled and state is updated on the flow model.
4. Flow model is added to the worklist collection via `this.collection.add(model)`, which makes it show in the UI.

Does the order of 3 & 4 matter? My assumptions is that either way the flow's state should show correctly in the UI.

Reply:
better to do data before render

Reply:
Yep 👍